go-controller/pkg/ovn/acl.go

```diff
@@ -0,0 +1,120 @@
+package ovn
+
+import (
+	"fmt"
+	"strings"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+
+	"k8s.io/klog/v2"
+)
+
+// GetNamespaceACLLogging retrieves ACLLoggingLevels for the Namespace.
+// nsInfo will be locked (and unlocked at the end) for given namespace if it exists.
+func (oc *Controller) GetNamespaceACLLogging(ns string) *ACLLoggingLevels {
+	nsInfo, nsUnlock := oc.getNamespaceLocked(ns, true)
+	if nsInfo == nil {
+		return &ACLLoggingLevels{
+			Allow: "",
+			Deny:  "",
+		}
+	}
+	defer nsUnlock()
+	return &nsInfo.aclLogging
+}
+
+func getLogSeverity(action string, aclLogging *ACLLoggingLevels) (log bool, severity string) {
+	severity = ""
+	if aclLogging != nil {
+		if action == nbdb.ACLActionAllow || action == nbdb.ACLActionAllowRelated || action == nbdb.ACLActionAllowStateless {
+			severity = aclLogging.Allow
+		} else if action == nbdb.ACLActionDrop || action == nbdb.ACLActionReject {
+			severity = aclLogging.Deny
+		}
+	}
+	log = severity != ""
+	return log, severity
+}
+
+// UpdateACLLoggingWithPredicate finds all ACLs based on a given predicate, updates log settings,
+// then transacts these changes with a single transaction.
+func UpdateACLLoggingWithPredicate(nbClient libovsdbclient.Client, p func(*nbdb.ACL) bool, aclLogging *ACLLoggingLevels) error {
+	ACLs, err := libovsdbops.FindACLsWithPredicate(nbClient, p)
+	if err != nil {
+		return fmt.Errorf("unable to list ACLs with predicate, err: %v", err)
+	}
+	return UpdateACLLogging(nbClient, ACLs, aclLogging)
+}
+
+func aclLoggingDisabled(aclName string) bool {
+	return strings.Contains(aclName, arpAllowPolicySuffix)
+}
```

Comment:
Now that I think of it we can't base our decision looking at the name. The names get trimmed to 64 chars and this is a suffix so won't work reliably for long names.

So what I would probably do is just use `buildDenyACLs` and compare them with the ones read from DB with `isEquivalentACL` and update logging on those required.

Reply:
now I think that if having these long names is something that really happens, then `isEquivalentACL` can be dangerous because having the same name and ExternalIDs will be enough to consider ACLs equal.
Then for defaultDenyACL, let's say namespace consumes the whole name, and the only ExternalIDs is `defaultDenyPolicyTypeACLExtIdKey: string(policyType)`, then arpAllowPolicy may be considered equivalent to the denyACL.
I still can forcefully compare by match,action,... but in general I think that if name truncation really happens, we will have more serious problems than loglevels

Reply:
You are right. In general I still think we should do as I said. `isEquivalentACL` is the right way to identify an ACL right now other than using the UUID.

If there is a bug in `isEquivalentACL` then:
* we might have to remove the name & external id check from `isEquivalentACL`. WE could do this right away and I think it would be fine but perhaps needs a scale check.
* or set an ExternalId['name'] that is not trimmed as we sync all the policies.

I guess we could use a different PR/BZ for that.